lessons/advanced/gen-stage-1.md

```diff
@@ -0,0 +1,368 @@
+---
```

Comment:
Okay. I think I have an idea to get this to fit into one and still hit on the points I was hoping for without getting too long-winded. I'll take a stab at that tomorrow, I have most of the code already done and have commits for the steps I'm shooting for but need to consider exactly how to go about it in writing things up + go into the details of what makes GenStage really work etc.

My main concern with how I have it now is being too hands-on and not really showing how it works under the hood, but I've got some ideas to alleviate that if expanding upon it.

Thanks!